util/reflect.go

```diff
@@ -46,3 +46,44 @@ func IsFieldUndefined(v reflect.Value) bool {
 		return false
 	}
 }
+
+// IsFieldPtr returns a boolean indicating whether the reflected value is a pointer.
+func IsFieldPtr(v reflect.Value) bool {
+	return v.Kind() == reflect.Ptr
+}
+
+// RecursivelySetUndefinedFields sets all fields that are not set in structToSet to the value of the corresponding field in structToDefaultFrom.
+// It takes two reflect.Values, the first is the struct to set the fields on, and the second is the struct to get the default values from.
```

Comment:
Thank you for the comment! This second sentence is probably not necessary since the variable names are explanatory but I'm not against it